src/test/java/org/gridsuite/securityanalysis/server/SecurityAnalysisControllerTest.java

```diff
@@ -244,14 +253,57 @@ public void stopTest() {
                 .expectBody(UUID.class)
                 .isEqualTo(RESULT_UUID);
 
+        Message<byte[]> message = output.receive(1000, "sa.run.destination");
+        assertEquals(NETWORK_UUID.toString(), message.getHeaders().get("networkUuid"));
+        assertEquals(RESULT_UUID.toString(), message.getHeaders().get("resultUuid"));
+        assertEquals(CONTINGENCY_LIST_NAME, message.getHeaders().get("contingencyListNames"));
+        assertEquals("me", message.getHeaders().get("receiver"));
+
         webTestClient.put()
                 .uri("/" + VERSION + "/results/" + RESULT_UUID + "/stop"
                         + "?receiver=me")
                 .exchange()
                 .expectStatus().isOk();
 
-        Message<byte[]> cancelMessage = output.receive(1000, "sa.cancel.destination");
-        assertEquals(RESULT_UUID.toString(), cancelMessage.getHeaders().get("resultUuid"));
-        assertEquals("me", cancelMessage.getHeaders().get("receiver"));
+        message = output.receive(1000, "sa.cancel.destination");
+        assertEquals(RESULT_UUID.toString(), message.getHeaders().get("resultUuid"));
+        assertEquals("me", message.getHeaders().get("receiver"));
+
+        message = output.receive(1000, "sa.stopped.destination");
+        assertEquals(RESULT_UUID.toString(), message.getHeaders().get("resultUuid"));
+        assertEquals("me", message.getHeaders().get("receiver"));
+
+        assertNull(output.receive(1000));
+    }
+
+    @Test
+    public void runTestWithError() {
+        webTestClient.post()
+                .uri("/" + VERSION + "/networks/" + NETWORK_UUID + "/run-and-save?contingencyListName=" + CONTINGENCY_LIST_ERROR_NAME
+                        + "&receiver=me")
+                .exchange()
+                .expectStatus().isOk()  // Because fully asynchronous (just publish a message)
+                .expectHeader().contentType(MediaType.APPLICATION_JSON)
+                .expectBody(UUID.class)
+                .isEqualTo(RESULT_UUID);
+
+        Message<byte[]> message = output.receive(1000, "sa.run.destination");
+        assertEquals(NETWORK_UUID.toString(), message.getHeaders().get("networkUuid"));
+        assertEquals(RESULT_UUID.toString(), message.getHeaders().get("resultUuid"));
+        assertEquals(CONTINGENCY_LIST_ERROR_NAME, message.getHeaders().get("contingencyListNames"));
+        assertEquals("me", message.getHeaders().get("receiver"));
+
+        // Message stopped has been sent
+        message = output.receive(1000, "sa.stopped.destination");
+        assertEquals(RESULT_UUID.toString(), message.getHeaders().get("resultUuid"));
+        assertEquals("me", message.getHeaders().get("receiver"));
+
+        assertNull(output.receive(1000));
+
+        // No result
+        webTestClient.get()
```

Comment:
should we do in this PR something to get the error here ?

Reply:
The error does not occur when this http request is executed because it only publish a message to launch AS (fully asynchronous) 
On the other hand we can recover the error in the stop message (see comment above)